charts/replicated-library/templates/lib/_ingress.tpl

```diff
@@ -47,10 +47,30 @@ spec:
       http:
         paths:
           {{- range .paths }}
-          {{- $service := $serviceName -}}
+          {{- $service := "" -}}
           {{- $port := 80 -}}
           {{- if .service -}}
-            {{- $service = default $service .service.name -}}
+          {{- range $key, $val := $.Values.services }}
+            {{- if and $val.enabled (eq $key $.ContextNames.ingress) -}}
+                 {{- $service = $.ContextNames.ingress -}}
+                 {{- $service = printf "%s-%s" (include "replicated-library.names.prefix" $) $service | trunc 63 | trimAll "-"  -}}
+            {{- end }}
+          {{- end }}
+          {{- if $values.serviceName -}}
+            {{- $service = $values.serviceName -}}
+             {{- $service = printf "%s-%s" (include "replicated-library.names.prefix" $) $service | trunc 63 | trimAll "-"  -}}
+          {{- end -}}
+          {{- if .service.name -}}
+              {{- $service = .service.name }}
+              {{- range $key, $val := $.Values.services }}
+                  {{- if and $val.enabled (eq $key $service) -}}
+                    {{- $service = printf "%s-%s" (include "replicated-library.names.prefix" $) $service | trunc 63 | trimAll "-"  -}}
+                  {{- end }}
+               {{- end }}
+            {{- end }}
+          {{- end }}
```

Comment:
```suggestion
          {{- if .service.name }}
              {{- $service = .service.name }}
           {{- else if $values.serviceName }}
               {{- $service = $values.serviceName }}
           {{- else }}     
               {{- range $key, $val := $.Values.services }}
                   {{- if and $val.enabled (eq $key $.ContextNames.ingress) }}
                       {{- $service = $.ContextNames.ingress }}
                   {{- end }}
           {{- end }}
           {{- range $key, $val := $.Values.services }}
               {{- if and $val.enabled (eq $key $service) -}}
                   {{- $service = printf "%s-%s" (include "replicated-library.names.prefix" $) $service | trunc 63 | trimAll "-"  -}}
               {{- end }}
           {{- end }}
      {{- end }}
```


I _think_ this does the same thing all I've done is rearrange the logic for readability/maintainability:
* Use a single if / else set to make it clear which names are used first.
* Use a single check for the prefix to keep it DRY

I haven't tested this, I think the indention and end statements line up.